relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -0,0 +1,235 @@
+#[cfg(feature = "processing")]
+use std::collections::{BTreeMap, BTreeSet};
+
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "processing")]
+use {
+    relay_common::UnixTimestamp,
+    relay_general::protocol::{AsPair, Event, EventType},
+    relay_metrics::{Metric, MetricUnit, MetricValue},
+};
+
+/// Configuration in relation to extracting metrics from transaction events.
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+#[serde(default, rename_all = "camelCase")]
+pub struct TransactionMetricsConfig {
+    extract_metrics: BTreeSet<String>,
+    extract_custom_tags: BTreeSet<String>,
+}
+
+#[cfg(feature = "processing")]
+pub fn extract_transaction_metrics(
+    config: &TransactionMetricsConfig,
+    event: &Event,
+    target: &mut Vec<Metric>,
+) {
+    if event.ty.value() != Some(&EventType::Transaction) {
+        return;
+    }
+
+    if config.extract_metrics.is_empty() {
+        return;
+    }
+
+    let mut push_metric = move |metric: Metric| {
+        if config.extract_metrics.contains(&metric.name) {
```

Comment:
nit: If the config check is the last thing we do for every metric, I would refactor to something like

```rust
extract_transaction_metrics(&event, &mut target); // no config!
target.retain(|x| config.extract_metrics.contains(x));
```

Although that opens the question how to handle tags. So probably this is fine the way it is.

Reply:
i want to early-return if there's no metrics to extract too, i think if i have the config handling in this function i can also unit-test it better